server/src/test/java/io/spine/examples/shareaware/server/e2e/given/WithClient.java

```diff
@@ -0,0 +1,102 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.e2e.given;
+
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.spine.client.Client;
+import io.spine.examples.shareaware.server.TradingContext;
+import io.spine.examples.shareaware.server.market.MarketDataProvider;
+import io.spine.server.Server;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.io.IOException;
+import java.time.Duration;
+
+import static io.grpc.ManagedChannelBuilder.forAddress;
+import static io.grpc.Status.CANCELLED;
+import static io.spine.client.Client.usingChannel;
+import static io.spine.server.Server.atPort;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Abstract base for end-to-end tests that aware of gRPC client.
+ */
+public abstract class WithClient {
+
+    private static final String ADDRESS = "localhost";
+    private static final int PORT = 4242;
+    private Client client;
+    private Server server;
+    private ManagedChannel channel;
+    private static final MarketDataProvider provider = MarketDataProvider.instance();
+
+    @BeforeAll
+    static void startProvider() {
+        provider.runWith(Duration.ofSeconds(1));
+    }
+
+    @AfterAll
+    static void stopProvider() {
+        provider.stopEmission();
+    }
+
+    @BeforeEach
+    void startAndConnect() throws IOException {
+        channel = forAddress(ADDRESS, PORT)
+                .usePlaintext()
+                .build();
+        server = atPort(PORT)
+                .add(TradingContext.newBuilder())
+                .build();
+        server.start();
+        client = usingChannel(channel).build();
```

Comment:
It might be better to move the `channel` opening and `client` creation to `E2ETestUser`. It will be necessary for tests with more than one user.

Reply:
I agree.